apps/emqx/include/http_api.hrl

```diff
@@ -0,0 +1,74 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2017-2022 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+%% Bad Request
+-define(BAD_REQUEST,              'BAD_REQUEST').
+
+-define(ALREADY_EXISTED,          'ALREADY_EXISTED').
+-define(BAD_CONFIG_SCHEMA,        'BAD_CONFIG_SCHEMA').
+-define(BAD_LISTENER_ID,          'BAD_LISTENER_ID').
+-define(BAD_NODE_NAME,            'BAD_NODE_NAME').
+-define(BAD_RPC,                  'BAD_RPC').
+-define(BAD_TOPIC,                'BAD_TOPIC').
+-define(EXCEED_LIMIT,             'EXCEED_LIMIT').
+-define(INVALID_PARAMETER,        'INVALID_PARAMETER').
+-define(CONFLICT,                 'CONFLICT').
+-define(NO_DEFAULT_VALUE,         'NO_DEFAULT_VALUE').
+-define(MESSAGE_ID_SCHEMA_ERROR,  'MESSAGE_ID_SCHEMA_ERROR').
+
+%% Resource Not Found
+-define(NOT_FOUND,                'NOT_FOUND').
+-define(CLIENTID_NOT_FOUND,       'CLIENTID_NOT_FOUND').
+-define(CLIENT_NOT_FOUND,         'CLIENT_NOT_FOUND').
+-define(MESSAGE_ID_NOT_FOUND,     'MESSAGE_ID_NOT_FOUND').
+-define(RESOURCE_NOT_FOUND,       'RESOURCE_NOT_FOUND').
+-define(TOPIC_NOT_FOUND,          'TOPIC_NOT_FOUND').
+-define(USER_NOT_FOUND,           'USER_NOT_FOUND').
+
+%% Internal error
+-define(INTERNAL_ERROR,           'INTERNAL_ERROR').
+-define(SOURCE_ERROR,             'SOURCE_ERROR').
+-define(UPDATE_FAILED,            'UPDATE_FAILED').
+-define(REST_FAILED,              'REST_FAILED').
+-define(CLIENT_NOT_RESPONSE,      'CLIENT_NOT_RESPONSE').
+
+%% All codes
+-define(ERROR_CODES, [
+    ?BAD_REQUEST,
+    ?ALREADY_EXISTED,
+    ?BAD_CONFIG_SCHEMA,
+    ?BAD_LISTENER_ID,
+    ?BAD_NODE_NAME,
+    ?BAD_RPC,
+    ?BAD_TOPIC,
+    ?EXCEED_LIMIT,
+    ?INVALID_PARAMETER,
+    ?CONFLICT,
+    ?NO_DEFAULT_VALUE,
+    ?MESSAGE_ID_SCHEMA_ERROR,
+    ?MESSAGE_ID_NOT_FOUND,
+    ?NOT_FOUND,
+    ?CLIENTID_NOT_FOUND,
+    ?CLIENT_NOT_FOUND,
+    ?RESOURCE_NOT_FOUND,
+    ?TOPIC_NOT_FOUND,
+    ?USER_NOT_FOUND,
+    ?INTERNAL_ERROR,
+    ?SOURCE_ERROR,
+    ?UPDATE_FAILED,
+    ?REST_FAILED,
+    ?CLIENT_NOT_RESPONSE
+]).
```

Comment:
I would be easy to change if we defined the atom codes directly in the ?ERROR_CODES macro, like:

```
-define(ERROR_CODES, [
     'BAD_REQUEST',
     'ALREADY_EXISTED'
].
```